cpp/src/arrow/engine/substrait/util.cc

```diff
@@ -40,113 +40,15 @@ namespace arrow {
 
 namespace engine {
 
-namespace {
-
-/// \brief A SinkNodeConsumer specialized to output ExecBatches via PushGenerator
-class SubstraitSinkConsumer : public compute::SinkNodeConsumer {
```

Comment:
I remember, I am doing a similar cleanup (based on your suggestion) in a particular PR (surely not merged yet), but I cannot remember where, but it is better we get this cleaned up and actually this was one of your suggestions as I can remember.

Reply:
Yes, that sounds likely.  If you can remember the PR I can take another look.

Reply:
But probably it is better to merge on your work, it is in the simplest form.